### ISSUE TITLE
efficient parallel list-list solver

### ISSUE THREAD

**Issue by bertsky** [poke1024/pyalign#8]
In rapidfuzz there's a [cdist](https://maxbachmann.github.io/RapidFuzz/Usage/process.html#rapidfuzz.process.cdist) function that computes a matrix of alignment scores between each pair of two collections [in parallel](https://github.com/maxbachmann/RapidFuzz/blob/aa6a88fae4ab331d9c05831ec80af8306eb8b6cd/src/rapidfuzz/process_cpp.hpp#L476).

Is there something similar in pyalign, too?

**Comment by poke1024** [comment]
no, currently not.

**Comment by poke1024** [comment]
There's a so-called batch mode, though I need to remind myself how it works. It makes use of SIMD calls to speed up computing multiple alignments at the same time.

**Comment by bertsky** [comment]
Ok, so in order to utilise multiple cores, I therefore have to wrap the solver (batched or not) in some Pythonic multiprocessing paradigm, right? Since builtin `multithreading` suffers from GIL and `multiprocessing` restricts sharing data to pickle-able objects, is there any one you can recommend in particular? (Like builtin concurrent.futures, or Joblib, Ray, MPIRE?)

**Comment by poke1024** [comment]
Yes, exactly. In theory, `multithreading` should already give good concurrency, since pyalign will release the GIL once it is inside the C++ code and does the heavy computations - I have never tested this out though. I have no suggestion to multiprocessing libs, I only have used the builtin `multiprocessing` so far.
